src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractPullReplicationHandlerProcessorForRegisterHubAccess.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations.Replication;
+using Raven.Client.Exceptions;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Raven.Server.Documents.Handlers.Processors.Databases;
+using Raven.Server.ServerWide.Commands;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Replication
+{
+    internal abstract class AbstractPullReplicationHandlerProcessorForRegisterHubAccess<TRequestHandler> : AbstractHandlerProcessorForUpdateDatabaseConfiguration<BlittableJsonReaderObject, TRequestHandler>
+        where TRequestHandler : RequestHandler
+    {
+        private string _hubTaskName;
+        private PullReplicationDefinition _hubDefinition;
+
+        protected AbstractPullReplicationHandlerProcessorForRegisterHubAccess([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask<BlittableJsonReaderObject> GetConfigurationAsync(TransactionOperationContext context, string databaseName, AsyncBlittableJsonTextWriter writer)
+        {
+            _hubTaskName = RequestHandler.GetStringQueryString("name", true);
+
+            _hubDefinition = RequestHandler.Server.ServerStore.Cluster.ReadPullReplicationDefinition(databaseName, _hubTaskName, context);
+            if (_hubDefinition == null)
+            {
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                return null;
+            }
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (_hubDefinition.Certificates != null && _hubDefinition.Certificates.Count > 0)
+#pragma warning restore CS0618 // Type or member is obsolete
+            {
+                // handle backward compatibility
+                throw new InvalidOperationException("Cannot register hub access to a replication hub that already has inline certificates: " + _hubTaskName +
+                                                    ". Create a new replication hub and try again");
+            }
+
+            return await context.ReadForMemoryAsync(RequestHandler.RequestBodyStream(), "register-hub-access");
+        }
+
+        protected override void OnBeforeUpdateConfiguration(ref BlittableJsonReaderObject configuration, JsonOperationContext context)
```

Comment:
there is a different method for assertion that should be used?

Reply:
done